sys/shell/shell.c

```diff
@@ -343,6 +375,9 @@ int shell_handle_input_line(const shell_command_t *command_list, char *line)
             print_help(command_list);
             return 0;
         }
+        else if (IS_USED(MODULE_SHELL_JSON) && !strcmp("cmds_json", argv[0])) {
+            print_commands_json(command_list);
+        }
```

Comment:
The issue is that the need to access the first parameter of `shell_run_forever()`. Externally implemented ones would only be able to list XFA based shell commands without accesa to that argument.

Reply:
Oh, I see! So it's an interesting mix of statically available xfa shell command definitions and a dynamically passeable command list at runtime via `shell_run_forever` (and friends?).

Can you think of use cases where it is actually necessary to have the dynamic command list? Otherwise we could store it upon call to `shell_run_forever` as a global state.

Reply:
I guess that would probably be fine. I think eventually the use of non-XFA shell commands will pass away over time.

Reply:
But maybe something for another PR, I guess :+1: